docs/software/packages/pytorch.md

```diff
@@ -0,0 +1,99 @@
+
+PyTorch can be installed by the users following the code's [instructions](https://pytorch.org/get-started/locally/). The options to choose for LUMI in the interactive instructions are `Linux`, `Pip` and `ROCm5.X`. For installing with `pip`, the `cray-python` module should be loaded. PyTorch comes with ROCm binaries needed for the GPU support. Even if a particular version of ROCm is not avilable on LUMI, PyTorch may still be able to use the GPUs.
+
+PyTorch can be run within containers as well. In particular, containers from the images provided by [AMD on DockerHub](https://hub.docker.com/u/rocm).
+Those images are updated frequently and make possible to try PyTorch with recent ROCm versions. 
+Another point in favor of using containers, is that PyTorch's installation directory can be quite large both in terms of storage size and number of files.
+
+## Running PyTorch within containers
+
+We recommend using container images from [`rocm/pytorch`](https://hub.docker.com/r/rocm/pytorch) or [`rocm/deepspeed`](https://hub.docker.com/r/rocm/deepspeed).
+
+The images can be fetched with singularity:
+```bash
+SINGULARITY_TMPDIR=$SCRATCH/tmp-singularity singularity pull docker://rocm/pytorch:rocm5.4.1_ubuntu20.04_py3.7_pytorch_1.12.1
+```
+This will create an image file named `pytorch_rocm5.4.1_ubuntu20.04_py3.7_pytorch_1.12.1.sif` on the directory where the command was run. After the image has been pulled, the directory `$SCRATCH/tmp-singularity singularity` can be removed.
+
+### Installing other packages along the container's PyTorch installation
+
+Often we may need to install other packages to be used along PyTorch.
+That can be done by creating a virtual environment within the container in a host directory.
+This can be done by running the container interactively and creating a virtual environment in your `$HOME`.
+As an example, let's do that to install the package `python-hostlist`:
+```bash
+$> singularity exec -B $SCRATCH:$SCRATCH pytorch_rocm5.4.1_ubuntu20.04_py3.7_pytorch_1.12.1.sif bash
+Singularity> python -m venv pt_rocm5.4.1_env --system-site-packages
+Singularity> . pt_rocm5.4.1_env/bin/activate
+(pt_rocm5.4.1_env) Singularity> pip install python-hostlist
+```
+Now when running the container, the virtual environment must be activated before calling python.
+
+## Multi-GPU training
+
+The communication between LUMI's GPUs during training with Pytorch is done via [RCCL](https://github.com/ROCmSoftwarePlatform/rccl), which is a library of  collective communication routines for GPUs. RCCL works out of the box on LUMI's, however, a special plugin is required so it can take advantadge of the [Slingshot interconnect](https://www.hpe.com/emea_europe/en/compute/hpc/slingshot-interconnect.html). That's the [`aws-ofi-rccl`](https://github.com/ROCmSoftwarePlatform/aws-ofi-rccl) plugin, which is library that can be used as backend for RCCL to interact with the interconnect via libfabric.
```

Comment:
typo

```suggestion
The communication between LUMI's GPUs during training with Pytorch is done via [RCCL](https://github.com/ROCmSoftwarePlatform/rccl), which is a library of  collective communication routines for GPUs. RCCL works out of the box on LUMI's, however, a special plugin is required so it can take advantage of the [Slingshot interconnect](https://www.hpe.com/emea_europe/en/compute/hpc/slingshot-interconnect.html). That's the [`aws-ofi-rccl`](https://github.com/ROCmSoftwarePlatform/aws-ofi-rccl) plugin, which is a library that can be used as a back-end for RCCL to interact with the interconnect via libfabric.
```